app/controllers/main/routes.py

```diff
@@ -66,7 +67,8 @@ def viewVolunteersProfile(username):
     try:
         volunteer = User.get(User.username == username)
     except Exception as e:
-        return "User does not exist", 404
+        flash(f"{username} does not exist! ", category='danger')
+        return redirect(url_for('admin.studentSearchPage'))
```

Comment:
I like this change!

One thing we need to check for is if a user has permissions to the search page before redirecting them there. The search page does check, but typing in an invalid username in the url and then getting redirected to a page you don't even have permission to would be weird. In that case, a 404 on the profile page makes more sense. For now, we can just use `isAdmin` on the current user.